chart/skywalking/templates/satellite-deployment.yaml

```diff
@@ -0,0 +1,133 @@
+      labels:
+        app: {{ template "skywalking.name" . }}
+        component: "{{ .Values.satellite.name }}"
+        release: {{ .Release.Name }}
+        {{- if .Values.satellite.podAnnotations }}
+      annotations:
+{{ toYaml .Values.satellite.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "skywalking.serviceAccountName.satellite" . }}
+      affinity:
+      {{- if eq .Values.satellite.antiAffinity "hard" }}
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app: "{{ template "skywalking.name" . }}"
+                  release: "{{ .Release.Name }}"
+                  component: "{{ .Values.satellite.name }}"
+      {{- else if eq .Values.satellite.antiAffinity "soft" }}
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            podAffinityTerm:
+              topologyKey: kubernetes.io/hostname
+              labelSelector:
+                matchLabels:
+                  app: "{{ template "skywalking.name" . }}"
+                  release: "{{ .Release.Name }}"
+                  component: "{{ .Values.satellite.name }}"
+      {{- end }}
+      {{- with .Values.satellite.nodeAffinity }}
+        nodeAffinity:
+{{ toYaml . | indent 10 }}
+      {{- end }}
+{{- if .Values.satellite.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.satellite.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.satellite.tolerations }}
+      tolerations:
+{{ toYaml .Values.satellite.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.imagePullSecrets | indent 8 }}
+{{- end }}
+      containers:
+      - name: {{ .Values.satellite.name }}
+        image: {{ .Values.satellite.image.repository }}:{{ required "satellite.image.tag is required" .Values.satellite.image.tag }}
+        imagePullPolicy: {{ .Values.satellite.image.pullPolicy }}
+        livenessProbe:
+          tcpSocket:
+            port: 11800
+          initialDelaySeconds: 15
+          periodSeconds: 20
+        ports:
+        {{- range $key, $value :=  .Values.satellite.ports }}
+        - containerPort: {{ $value }}
+          name: {{ $key }}
+        {{- end }}
+{{- if .Values.satellite.resources }}
+        resources:
+{{ toYaml .Values.satellite.resources | indent 10 }}
+{{- end }}
+        env:
+        - name: SATELLITE_GRPC_CLIENT_FINDER
+          value: kubernetes
+        - name: SATELLITE_GRPC_CLIENT_KUBERNETES_NAMESPACE
+          value: "{{ .Release.Namespace }}"
+        - name: SATELLITE_GRPC_CLIENT_KUBERNETES_KIND
+          value: pod
+        - name: SATELLITE_GRPC_CLIENT_KUBERNETES_SELECTOR_LABEL
+          value: "app={{ template "skywalking.name" . }},release={{ .Release.Name }},component={{ .Values.oap.name }}"
```

Comment:
Is `component` not `{{ .Values.satellite.name }}`?

Reply:
yes, the satellite needs to get real oap ports, not itself.

Reply:
got your point. it's an oap pod discovery mechanism. 

Could you add a helper function to wrap them up since both oap and this are referring them ？